### ISSUE TITLE
Fill out colophon page

### ISSUE THREAD

**Issue by thingles** [thingles/links.github.io#6]
Add information about how this site works and is constructed.

**Comment by nitinthewiz** [comment]
Yes please! I vote for this! Though I've mostly figured it out - 
1. Netlify your GitHub repo
2. ???
3. ???

Well, I guess I don't have everything figured out. Does Netlify run your make script frequently? Do you run the make script on a VPS or home computer, thus calling the pinboard API and generating your posts every time from scratch? That feels a little weird and unnecessary.

**Comment by thingles** [comment]
Hello @nitinthewiz! I gave [a reply on micro.blog](https://micro.blog/jthingelstad/842781) but figured I would also comment here. 

You are correct about the process, I run it on my laptop and blow away all old data and recreate it. That was easier than detecting changes in each of the posts, and I haven't spent any time on this script since hacking it into existence.

Pinboard has an API method to show the date/time of the last change, in essence to tell a third party if a sync is needed. My ultimate plan has been to have an AWS Lambda function run every 30 mins or so, poll that date/time, and if needed it could invoke another Lambda to update posts. That could then commit to Github, or, I could move the hosting to AWS and just have the last Lambda update and S3 bucket. Honestly I'd probably leave it with Github and Netlify though. I've also desired to move to Hugo since the build in Jekyll takes several minutes.

The important part is using Pinboard as a CMS. Put Markdown in the description, and I’m using Tags as metadata indicators. Currently I don't use that on my link blog, only my newsletter. But the "_feature" and "_brief" tags should indicate a different display. 

If you have any interest in extending it in these directions perhaps we could work together? Ultimately the thing that is neat is a Pinboard as CMS to Markdown files thing. Theme, wether it’s Hugo or Jekyll, is all after that.

**Comment by nitinthewiz** [comment]
@thingles thank you so much for the details! Your vision seems interesting, because AWS Lambda would really solve a lot of problems with the laptop approach.

Sadly, I'm not a pinboard user. I saw the way you make your posts and how you consume them (through the API) and immediately started thinking of creating files directly, using prose.io or something similar to write directly to github. Not the most elegant or reusable approach, but hey, that's where I was going.

I can work on the tags idea independently and see if I can come up with something. I've never played with either Hugo or Jekyll and your code seemed a very easy out-of-the-box solution for me, with a few tweaks, of course.

That said, I'm always looking at pinboard and wondering if I should kill all the other gazillion approaches and just go for pinboard 😆

**Comment by nitinthewiz** [comment]
Gosh, I just built your jekyll site on Windows. It took 517.925 seconds! That's some time frame! If you're saying Hugo takes less time, you should absolutely look at that (and so will I). After all, even if you go the AWS Lambda route, that'll cost good money too, over time!

**Comment by thingles** [comment]
@nitinthewiz yes, the tests I've done using Hugo are 20x faster. Just need to get the theme situation in hand.

**Comment by nitinthewiz** [comment]
Starting points?
-> https://github.com/chmking/hugo-solarized
-> https://github.com/BakeRolls/solar-theme-hugo

**Comment by nitinthewiz** [comment]
I've been thinking - Netlify supports Lambda - they give you their own interface and libraries and when you upload the code to them, they'll run the function on AWS Lambda for you.

So, would it make sense to do the following - 

1. For you - a Netlify lambda function that runs every time your pinboard account updates. This could be a cron-like job at Netlify's end, or if they don't allow that, a webpage that you hit every time you make an update to your pinboard account. The webpage will just call the Netlify Lambda function using JS and run in the background to execute your build. The number of times this can run for free is massive - 125,000 function requests per month. But the gating function is the number of hours it can run - 100 hours/mo. Based on that 500+ second build time, that maps to just about 700 runs/mo. That's a LOT, but that's a decision you have to make - go for Hugo and make this a faster process or not. Of course, if we make the code such that it only updates from your latest pinboard changes, that'll make the process **much** faster.

2. For me - I don't want to do pinboard. I want to go the route of having a bookmarklet that pops up a form. The form will take my URL, tags, comment, and quote, and run a Netlify Lambda function that uploads that all to Github as an md file, then builds the static site and uploads that to github as well, after which Netlify would automatically pick it up and host it.

What do you think?
